sympy/core/assumptions.py

```diff
@@ -196,13 +196,29 @@ class StdFactKB(FactKB):
     """
     rules = _assume_rules
 
-    def __init__(self, facts=None):
+    def __init__(self, facts=None, extra_facts=None):
```

Comment:
I don't think adding the `extra_facts` kwarg is ideal. It is definitely a hack.

Is there anything wrong with storing commutative all the time? It makes the srepr strings longer, but maybe that's what we should do until a better fix can be applied for dealing with the system adding commutative. I'd suggest storing all user applied assumptions plus commutative so the hack isn't needed. Then open an issue that says "don't store commutative assumption in srepr".